remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingClient.java

```diff
@@ -707,5 +711,25 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
                 NettyRemotingClient.this.putNettyEvent(new NettyEvent(NettyEventType.EXCEPTION, remoteAddress, ctx.channel()));
             }
         }
+
+        @Override
+        public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+            Channel channel = ctx.channel();
+            if (null == channel) {
+                return;
+            }
+            if (responseTable != null) {
+                Iterator<Map.Entry<Integer, ResponseFuture>> it = responseTable.entrySet().iterator();
+                while (it.hasNext()) {
+                    Map.Entry<Integer, ResponseFuture> entry = it.next();
+                    if (channel == entry.getValue().getProcessChannel()) {
+                        final String remoteAddress = RemotingHelper.parseChannelRemoteAddr(ctx.channel());
+                        log.warn("NETTY CLIENT PIPELINE: channelInactive, so wakeup response, the channel[{}]", remoteAddress);
+                        entry.getValue().wakeupResponse();
+                    }
+                }
+            }
+            super.channelInactive(ctx);
+        }
```

Comment:
In order to solve the issueeof timeout, is it worth scanning all items of responseTable?

Reply:
I have optimized the structure, migrate wakeup logic to NettyRemotingAbstract. Similar to NettyRemotingAbstract#failFast, scan is required.

Reply:
> In order to solve the issueeof timeout, is it worth scanning all items of responseTable?

I have optimized the structure, migrate wakeup logic to NettyRemotingAbstract. Similar to NettyRemotingAbstract#failFast, scan is required.